File: CS50x/PSet2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

// declare functions
int count_letters(string texto);
int count_words(string texto);
int count_sentences(string texto);


int main(void)
{
    // input a text
    string texto = get_string("Text: ");
    // declare variables
    float letras, palavras, frases;
    letras = count_letters(texto);
    palavras = count_words(texto);
    frases = count_sentences(texto);

    // calculating the indeex
    float L, S;
    double index;
    L = (letras / palavras * 100);
    S = (frases / palavras * 100);
    index = round(0.0588 * L - 0.296 * S - 15.8);

    // what to plot
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        if (index <= 0)
        {
            printf("Before Grade 1\n");

        }
        else
        {
            int int_index = (int) index;
            printf("Grade %i\n", int_index);
        }
    }

}
// funtion to sum the number of letters
int count_letters(string texto)
{
    int letras = 0;
    for (int i = 0; i < strlen(texto); i++)
    {
        if (isalpha(texto[i]))
        {
            letras += 1;
        }
    }
    // printf("%i letras\n", letras);
    return letras;
}

// funtion to sum the number of words
int count_words(string texto)
{
    int palavras = 1;
    for (int i = 0; i < strlen(texto); i++)
    {
        if (isspace(texto[i]))
        {
            palavras += 1;
        }
    }
    // printf("%i palavras\n", palavras);
    return palavras;
}

// funtion to sum the number of sentences
int count_sentences(string texto)
{
    int frases = 0;
    for (int i = 0; i < strlen(texto); i++)
    {
        if (texto[i] == '.' || texto[i] == '!' || texto[i] == '?')
        {
            frases += 1;
        }
    }
    // printf("%i frases\n", frases);
    return frases;
}

File: CS50x/PSet6/sentimental-readability/readability.py
from cs50 import get_string


text = get_string("Text: ")


def main(text):
    # call letters, words and sentences functions
    letters = numLetters(text)
    words = numWords(text)
    sentences = numSentences(text)

    # calculate index
    L = (letters / words * 100)
    S = (sentences / words * 100)
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # print grade
    if index >= 16:
        print("Grade 16+")
    elif index <= 0:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


# letter function
def numLetters(text):
    letters = 0
    for e in text:
        if e.isalpha() == True:
            letters += 1
    return letters


# words function
def numWords(text):
    listWords = text.split(' ')
    words = len(listWords)
    return words


# sentences function
def numSentences(text):
    sentences = 0
    for e in text:
        if e == "." or e == "?" or e == "!":
            sentences += 1
    return sentences


main(text)

